code/modules/clothing/clothing.dm

```diff
@@ -292,6 +297,80 @@
 			siemens_coefficient = between(0, material.conductivity / 10, 10)
 		slowdown = between(0, round(material.weight / 10, 0.1), 6)
 
+/obj/item/clothing/proc/get_accessory(var/typepath)
+	if(istype(src, typepath))
+		return src
+	if(LAZYLEN(accessories))
+		var/accessory = locate(typepath) in accessories
+		if(accessory)
+			return accessory
+	return null
+
+// Hood relevant parts. Look at /obj/item/clothing/suit/storage/hooded for how to implement the verb to use it
+/obj/item/clothing/proc/MakeHood()
+	var/obj/item/clothing/C = get_accessory(clothing_type)
+	if(!C.hood)
+		C.hood = new hoodtype(src)
+
+/obj/item/clothing/Destroy()
+	var/obj/item/clothing/C = get_accessory(clothing_type)
+	QDEL_NULL(C.hood)
```

Comment:
for instance, the hood here would listen to `COMSIG_PARENT_QDELETING` and qdel itself accordingly